fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/loanschedule/domain/AbstractLoanScheduleGenerator.java

```diff
@@ -178,7 +178,58 @@ private LoanScheduleModel generate(final MathContext mc, final LoanApplicationTe
                 isFirstRepayment = false;
             }
         }
-        while (!scheduleParams.getOutstandingBalance().isZero() || !scheduleParams.getDisburseDetailMap().isEmpty()) {
+
+        Money totalDownPaymentAmount = Money.zero(loanApplicationTerms.getCurrency());
+
+        if (loanApplicationTerms.isDownPaymentEnabled()) {
+            if (!scheduleParams.getDisburseDetailMap().isEmpty()) {
+                for (Map.Entry<LocalDate, Money> disburseDetail : scheduleParams.getDisburseDetailMap().entrySet()) {
```

Comment:
Should we check whether the disbursement was already disbursed? I might be wrong, but i dont we are storing the "expected" disbursements as well, but should it mean we need to create the down payment installment for them?

Reply:
refactored